presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/clientproviders/RetryingPrestoThriftServiceProvider.java

```diff
@@ -162,7 +174,6 @@ public void close()
             catch (Exception e) {
                 log.warn(e, "Error closing client");
             }
-            client = null;
```

Comment:
Don't remove this line, it's critical! Add a test to verify that we create a new client.